internal/metrics/remote_write.go

```diff
@@ -153,19 +160,58 @@ func (r *RemoteWrite) Update(config models.DeviceConfigurationMessage) error {
 			return fmt.Errorf("metrics remote write configuration is invalid. RequestNumSamples has to greater than 0")
 		}
 
+		// create TLS client config
+		clientConfig := config_util.HTTPClientConfig{}
+
+		if newConfig.CaCert != "" {
+			// create new file for CA. can't use same file cause client watches for changes in the file we pass to it.
+			caFile, err := ioutil.TempFile(r.dataDir, ServerCAFileNamePattern)
+			if err != nil {
+				log.Error("could not create temp file for metics remote write server CA")
+				return err
+			}
+
+			caFilePath := caFile.Name()
+
+			_, err = caFile.WriteString(newConfig.CaCert)
+			if err == nil {
+				err = caFile.Sync()
+				if err == nil {
+					err = caFile.Close()
+				}
+			}
+			if err != nil {
```

Comment:
Not sure about the `err` checks.
Does `if err != nil` refer to `_, err = caFile.WriteString(newConfig.CaCert)` ?
if yes, then this if block should be placed after `_, err = caFile.WriteString`
if no, (i.e. `err` refer to ` caFile.Close()`) then there isn't a check on `caFile.WriteString`.

Can you please clarify?

Reply:
instead of checking that error occurred and then log and return, I check that err did not occur and then continue to next operation. After all operations (WriteString, Sync, Close) I check whether occurred. I then log it and return

Reply:
IMO this "code style" complicates a bit the understating of the code since checking if `err != nil` is the idiomatic way to determine if an error was encountered.
But if you like it more this way, that's fine.

Reply:
In this case it saves duplicity of message and code. In many cases each error has different log msg and different error handling. Here it is the same for all cases. So i save code by doing it this way. It is not a 'method' i use.  It is specifc to this case